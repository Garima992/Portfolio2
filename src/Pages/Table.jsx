import * as React from 'react';

const Table=()=>{
    const data=[
        {id:1, name:'John Wynk', email:'john@example.com', phone:'777-777-7777'},
        {id:2, name:'Jane Lane', email:'jane@example.com', phone:'555-555-5555'},
        {id:3, name:'Bob Smith', email:'bobb@example.com', phone:'222-222-2222'},
        {id:4, name:'Alice Murray', email:'alice@example.com', phone:'111-111-1111'}
    ];


return(
        <div className='main-table'>
            <div className='text'>
                <h2>Hero</h2>
                <div id='name' data-testid="hero">hulk</div>
            </div>
            <table className="table">
                <thead>
                    <tr>
                        <th>ID</th>
                        <th>Name</th>
                        <th>Email</th>
                        <th>Phone</th>
                    </tr>
                </thead>
                <tbody>
                    {data.map((item)=>(
                            <tr>key={item.id}
                            <td>{item.id}</td>
                            <td>{item.name}</td>
                            <td>{item.email}</td>
                            <td>{item.phone}</td>
                            </tr> 
                    ))}
                </tbody>
            </table>
        </div>
    );
};

export default Table;