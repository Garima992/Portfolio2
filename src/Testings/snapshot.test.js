import * as React from 'react';
import Table from '../Pages/table';
import {cleanup, render, screen} from '@testing-Library/react';

afterEach(cleanup);

it('shouldloadand display the table', async()=>{
    const renderedSnapshot = render(<Table />)
    expect(renderedSnapshot.asFragment(<Table/>)).toMatchSnapshot();
});