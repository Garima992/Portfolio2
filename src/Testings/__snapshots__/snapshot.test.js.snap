// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shouldloadand display the table 1`] = `
<DocumentFragment>
  <div
    class="main-table"
  >
    <div
      class="text"
    >
      <h2>
        Hero
      </h2>
      <div
        data-testid="hero"
        id="name"
      >
        hulk
      </div>
    </div>
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            ID
          </th>
          <th>
            Name
          </th>
          <th>
            Email
          </th>
          <th>
            Phone
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            John Doe
          </td>
          <td>
            john@example.com
          </td>
          <td>
            777-777-7777
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            Jane Lane
          </td>
          <td>
            jane@example.com
          </td>
          <td>
            555-555-5555
          </td>
        </tr>
        <tr>
          <td>
            3
          </td>
          <td>
            Bob Smith
          </td>
          <td>
            bobb@example.com
          </td>
          <td>
            222-222-2222
          </td>
        </tr>
        <tr>
          <td>
            4
          </td>
          <td>
            Alice Murray
          </td>
          <td>
            alice@example.com
          </td>
          <td>
            111-111-1111
          </td>
        </tr>
      </tbody>
    </table>
    key=1
    key=2
    key=3
    key=4
  </div>
</DocumentFragment>
`;
